numba/tests/test_record_dtype.py

```diff
@@ -1495,6 +1513,20 @@ def test_set_arrays(self):
             arr_res = cfunc(nbarr)
             np.testing.assert_equal(arr_res, arr_expected)
 
+    def test_setitem(self):
+        nbarr1 = np.recarray(2, dtype=recordwith2darray)
```

Comment:
Why are these size 2 if only the first index is assigned below, should they be size 1?

Reply:
Now resolved using your fix.